### ISSUE TITLE
[FP] Suspicious binary operations: bit operations

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1227]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 2.0.13 |

Current behaviour
--
There a few case of using & for bit operations that are legitimate and we could avoid reporting:

1. With any usage with hexadecimal notation (0x...) we could assume the developper know what he is doing. Example:
```php
$var = $var2&0xFFFF;
```

2. With any usage with octal notation, we could assume the developper know what he is doing. Example:
```php
$var = 0000001&0000010;
```

3. When there is a bit wise operation with any int which is not 0. In any case, the suggestion to replace by an && is irrelevant, because an int is always true so it makes no sense to check its boolean value. Example:
```php
$var = 10&1;
```

4. Same case as 3. but with constant instead. If the constant is an int which is not 0.
Example:
```php
$var = FLAG1&FLAG2;
```
I guess this could be applied to class constant as well.

**Comment by orklah** [comment]
I forgot binary notation:

```php
$var = 0b11111111&0b0000000
```

**Comment by orklah** [comment]
Please note: I used "&" for all my examples, but they are also valid with "|".

While trying to write examples, I thought about an inspections that could check for always true cases/ always false cases with bit operations.
This is not very much used so I don't know if it's worth it. Tell me if it seems interesting, I could try to write an issue with all the details

**Comment by kalessil** [comment]
Fixed! I also considered the constant conditions case earlier but decided to not implement it.
